cpp/src/streaming/core/spillable_messages.cpp

```diff
@@ -26,6 +25,26 @@ Message SpillableMessages::extract(MessageId mid) {
     return std::exchange(item->message, std::nullopt).value();
 }
 
+Message SpillableMessages::copy(MessageId mid, BufferResource* br) const {
+    std::unique_lock global_lock(global_mutex_);
+    auto it = items_.find(mid);
+    RAPIDSMPF_EXPECTS(
+        it != items_.end(), "message not found " + std::to_string(mid), std::out_of_range
+    );
+    std::shared_ptr<Item> item = it->second;
+    global_lock.unlock();
+
+    std::unique_lock item_lock(item->mutex);
+    RAPIDSMPF_EXPECTS(
+        item->message.has_value(),
+        "empty message " + std::to_string(mid),
+        std::out_of_range
+    );
+    auto const& msg = *item->message;
+    auto res = br->reserve_or_fail(msg.copy_cost(), msg.spillable_memory_types());
+    return msg.copy(res);
```

Comment:
I'm not 100% sure if this is good because every time we make a copy, we would have to lock-unlock two mutexes. 
Can we keep a copy in the fanout and make copies there, rather than copying messages from spillable messages? @madsbk WDYT?